spacy_layout/layout.py

```diff
@@ -112,19 +124,42 @@ def _texts_to_doc(self, inputs: list[tuple[str, str, SpanLayout]]) -> Doc:
                     spaces[-1] = False
                     spaces.append(False)
                 end = token_idx + len(span_doc)
-                span_data.append((token_idx, end, label, layout))
+                span_data.append((item, token_idx, end))
                 token_idx += len(span_doc) + (1 if self.sep else 0)
         doc = Doc(self.nlp.vocab, words=words, spaces=spaces)
         spans = []
-        for i, (start, end, label, layout) in enumerate(span_data):
-            span = Span(doc, start=start, end=end, label=label, span_id=i)
+        for i, (item, start, end) in enumerate(span_data):
+            span = Span(doc, start=start, end=end, label=item.label, span_id=i)
+            layout = self._get_span_layout(item, pages)
             span._.set(self.attrs.span_layout, layout)
+            if item.label == DocItemLabel.TABLE:
+                span._.set(self.attrs.span_data, item.export_to_dataframe())
             spans.append(span)
         doc.spans[self.attrs.span_group] = SpanGroup(
             doc, name=self.attrs.span_group, spans=spans
         )
         return doc
 
+    def _get_span_layout(
+        self, item: DoclingItem, pages: dict[int, PageLayout]
+    ) -> SpanLayout | None:
+        bounding_box = None
+        if item.prov:
+            prov = item.prov[0]
+            page = pages[prov.page_no]
+            if page.width and page.height:
+                is_bottom = prov.bbox.coord_origin == CoordOrigin.BOTTOMLEFT
+                y = page.height - prov.bbox.t if is_bottom else prov.bbox.t
+                height = prov.bbox.t - prov.bbox.b if is_bottom else prov.bbox.t
+                bounding_box = SpanLayout(
+                    x=prov.bbox.l,
+                    y=y,
+                    width=prov.bbox.r - prov.bbox.l,
+                    height=height,
+                    page_no=prov.page_no,
+                )
+        return bounding_box
+
     def get_pages(self, doc: Doc) -> list[tuple[PageLayout, list[Span]]]:
```

Comment:
Is there a way to easily navigate to a `PageLayout` object from the page-span?

I think there's some advantage to having things related through the object access, so that the user's data structures are simpler. They can work with a `Span` object instead of the tuple.

Reply:
The `span._.layout` gives you the page number, so you can do `doc._.pages[span._.layout.page_no]`. Maybe it actually give you the page instead, which is a good idea.